File: benchmarks/fair/fib.py
import os, sys
sys.setrecursionlimit(100000)
def fib(n):
    return n if n <= 1 else fib(n-1) + fib(n-2)
n = int(os.environ.get("BENCH_N") or 0) or 32
print(fib(n))

File: benchmarks/fair/fib.cpp
#include <cstdio>
#include <cstdlib>
static long long fib(long long n){ return n<=1?n:fib(n-1)+fib(n-2); }
int main(){
    const char*e=std::getenv("BENCH_N"); long long n=e?std::atoll(e):0; if(n<=0)n=32;
    std::printf("%lld\n",fib(n)); return 0;
}
